src/app/loadout-builder/filter/ModPicker.tsx

```diff
@@ -15,33 +15,36 @@ import { escapeRegExp } from 'app/search/search-filters/freeform';
 import { SearchFilterRef } from 'app/search/SearchBar';
 import { AppIcon, searchIcon } from 'app/shell/icons';
 import { RootState } from 'app/store/types';
-import { chainComparator, compareBy } from 'app/utils/comparators';
 import { isArmor2Mod } from 'app/utils/item-utils';
 import { DestinyClass, DestinyProfileResponse } from 'bungie-api-ts/destiny2';
 import _ from 'lodash';
-import React, { Dispatch, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 import Sheet from '../../dim-ui/Sheet';
 import '../../item-picker/ItemPicker.scss';
-import { LoadoutBuilderAction } from '../loadout-builder-reducer';
-import { knownModPlugCategoryHashes, LockedMods } from '../types';
+import { sortModGroups, sortMods } from '../mod-utils';
+import { PluggableItemsByPlugCategoryHash } from '../types';
 import { isLoadoutBuilderItem } from '../utils';
 import ModPickerFooter from './ModPickerFooter';
 import PickerSectionMods from './PickerSectionMods';
 
-// to-do: separate mod name from its "enhanced"ness, maybe with d2ai? so they can be grouped better
-const sortMods = chainComparator<PluggableInventoryItemDefinition>(
-  compareBy((mod) => mod.plug.energyCost?.energyType),
-  compareBy((mod) => mod.plug.energyCost?.energyCost),
-  compareBy((mod) => mod.displayProperties.name)
-);
-
 interface ProvidedProps {
-  lockedMods: LockedMods;
+  /**
+   * An object of plugCatgeoryHashes to mods (PluggableInventoryItemDefinition[])
+   * with that plugCategoryHash.
+   */
+  lockedMods: PluggableItemsByPlugCategoryHash;
```

Comment:
Why have the consumer of this component group these instead of grouping them internally?

Reply:
They are grouped as they are indexed by plugCategoryHash but they aren't ordered yet. I felt like ordering was a rendering problem so I pushed it as close to rendering as I could. Also once ordered its an array or arrays which I normally avoid unless necessary. I could pull it higher though if you prefer, I am not too fussed on that part.

Reply:
What I meant was, why not have `LoadoutBuilder` pass a flat list of locked mods, and do the indexing by `plugCategoryHash` inside `ModPicker`?

Reply:
I am planning on doing something like this but it involves changing the lockedmods state in the optimiser. Basically I want to get rid of the whole LockedMod/s type and just run it off a list of PluggableInventoryItemDefinition's. I felt it was out of scope for this PR though. I will do it in a follow up one.